scripts/jenkins/ardana/ansible/roles/heat_stack/library/ecp_os_stack.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# (c) 2016, Mathieu Bultel <mbultel@redhat.com>
+# (c) 2016, Steve Baker <sbaker@redhat.com>
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+import time
+
+from ansible.module_utils._text import to_native
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.openstack import openstack_cloud_from_module, \
+    openstack_full_argument_spec, openstack_module_kwargs
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: ecp_os_stack
```

Comment:
How is this different from the module it was forked from?
Why is there a need to have a copy here?

Reply:
The only difference is, due to the ECP API not being reliable (sometimes when creating a stack it returns HTTP 500), I have added a retry when the request returns 500.

Reply:
Please in a commit above the copyright claim add a link containing the commit hash to the upstream used as base. I didn't check, but maybe it is:
https://github.com/ansible/ansible/blob/a8d4bf86421d151d8df7132e8e87d04b6662f45a/lib/ansible/modules/cloud/openstack/os_stack.py

Do you mind if someone else submits the changes upstream?

Reply:
To get the codeclimate check to not fail this, you also need to exclude this file.
Add a file named `.codeclimate.yml` to the top of the repo with the following content:
```
exclude_patterns:
- "scripts/jenkins/ardana/ansible/roles/heat_stack/library/ecp_os_stack.py "
```

Reply:
> Please in a commit above the copyright claim add a link containing the commit hash to the upstream used as base. I didn't check, but maybe it is:
> [ansible/ansible:lib/ansible/modules/cloud/openstack/os_stack.py@`a8d4bf8`](https://github.com/ansible/ansible/blob/a8d4bf86421d151d8df7132e8e87d04b6662f45a/lib/ansible/modules/cloud/openstack/os_stack.py)
> 
> Do you mind if someone else submits the changes upstream?

no.


> To get the codeclimate check to not fail this, you also need to exclude this file.
> Add a file named `.codeclimate.yml` to the top of the repo with the following content:
> 
> ```
> exclude_patterns:
> - "scripts/jenkins/ardana/ansible/roles/heat_stack/library/ecp_os_stack.py "
> ```

I have added the file, thanks did not know about it :)